pysph/examples/rb_rotation_matrices/stack_of_cylinders.py

```diff
@@ -0,0 +1,239 @@
+            x=xw, y=yw, h=h, m=m, rho=self.wall_rho,
+            rad_s=self.wall_spacing / 2., V=self.wall_spacing**2, name="wall")
+        return [cylinders, dam, wall]
+
+    def create_solver(self):
+        kernel = CubicSpline(dim=2)
+
+        integrator = EPECIntegrator(cylinders=RK2StepRigidBody())
+
+        dt = self.dt
+        print("DT: %s" % dt)
+        tf = self.tf
+        solver = Solver(kernel=kernel, dim=2, integrator=integrator, dt=dt,
+                        tf=tf)
+
+        return solver
+
+    def create_equations(self):
+        equations = [
+            Group(
+                equations=[
+                    BodyForce(dest='cylinders', sources=None, gy=-9.81),
+                ], real=False),
+            Group(equations=[
+                RigidBodyCollision(dest='cylinders', sources=[
+                    'dam', 'wall', 'cylinders'
+                ], kn=1e7, en=0.5),
+            ]),
+            Group(equations=[
+                SumUpExternalForces(dest='cylinders', sources=None)
+            ]),
+        ]
+        return equations
+
+    def create_dam(self):
+        xt, yt = get_2d_tank(self.dam_spacing,
+                             np.array([self.dam_length / 2., 0.]),
+                             length=self.dam_length, height=self.dam_height,
+                             num_layers=self.dam_layers, outside=True)
+
+        return xt, yt
+
+    def create_wall(self):
+        x = np.arange(0.067, 0.067 + 2 * self.wall_spacing, self.wall_spacing)
+        y = np.arange(0., self.wall_height, self.wall_spacing)
+        xw, yw = np.meshgrid(x, y)
+        return xw.ravel(), yw.ravel()
+
+    def create_cylinders_stack(self):
+        # create a stack of cylinders
+
+        # first crate all six cylinder rows
+        # first six set of cylinders
+        xc1, yc1 = get_2d_circle(self.cylinder_spacing, self.cylinder_radius, [
+            self.cylinder_radius + self.cylinder_spacing,
+            self.cylinder_radius + self.cylinder_spacing
+        ])
+
+        xc2, yc2 = (xc1 + self.cylinder_diameter + self.cylinder_spacing, yc1)
+        xc3, yc3 = (xc2 + self.cylinder_diameter + self.cylinder_spacing, yc2)
+        xc4, yc4 = (xc3 + self.cylinder_diameter + self.cylinder_spacing, yc3)
+        xc5, yc5 = (xc4 + self.cylinder_diameter + self.cylinder_spacing, yc4)
+        xc6, yc6 = (xc5 + self.cylinder_diameter + self.cylinder_spacing, yc5)
+
+        x_six_1, y_six_1 = np.concatenate(
+            (xc1, xc2, xc3, xc4, xc5, xc6)), np.concatenate((yc1, yc2, yc3,
+                                                             yc4, yc5, yc6))
+        x_six_2, y_six_2 = x_six_1, y_six_1 + 2. * self.cylinder_diameter
+        x_six_3, y_six_3 = x_six_2, y_six_2 + 2. * self.cylinder_diameter
+
+        x_six, y_six = np.concatenate(
+            (x_six_1, x_six_2, x_six_3)), np.concatenate((y_six_1, y_six_2,
+                                                          y_six_3))
```

Comment:
Have you though of doing this?
    
    r = 0.5
    px = 0.5 + 1.0 # pitch along x
    py = 2 * 2 * r # pitch along y
    x = []
    y = []
    for i in range(6):
        for j in range(5):
                c0, c1 = px * i, py * j
                xc1, yc1 = get_2d_circle(dx, r, [c0, c1])
                x = np.concatenate(x, xc)
                y = np.concatenate(y, yc)

Reply:
I agree, why haven't you done this?  Much easier to read and less messy.